os/board/rtl8721csm/src/component/soc/realtek/amebad/fwlib/ram_hp/rtl8721dhp_app_start.c

```diff
@@ -1321,7 +1321,6 @@ void app_start(void)
 	}
 
 #ifndef CONFIG_PLATFORM_TIZENRT_OS
-	os_heap_init();
```

Comment:
You should not remove this. You should move this to be built and to work.

Reply:
Done. Thanks.

Reply:
Since we are checking below condition, we dont need to remove this.

Reply:
Why do we need os_heap_init() for Tizenrt?
It is modifying our kregionx_xxx array which is generated from our config file. 
If this api is called, then our changes for 3 regions and 2 heaps will not work.

Reply:
I think it is okay.  Our mm_initialize and addregion will work based on modified address and size, because app_start is called before os_start.

Reply:
Our kregion has wrong values, because Realtek adds some text, bss, and data into PSRAM.
```
	/* PSRAM_NS image start */
	.psram_image2.text :
	{
		__psram_image2_text_start__ = .;
		*(.psram.text*)
		__psram_image2_text_end__ = .;
	} > PSRAM_NS

	.psram_image2.data :
	{
		. = ALIGN (32);
		*(.psram.data*)
		. = ALIGN (32);
		*(.psram.rodata*)
	} > PSRAM_NS

	.psram_image2.bss :
	{
		. = ALIGN (32);
		__psram_bss_start__ = .;
		*(.psram.bss*)
		__psram_bss_end__ = .;
	} > PSRAM_NS
```
So we can use PSRAM for heap after that region. But our config does not concern that.
That would be checked by PR #4550 which was modifying by Sangam.
Before that, we should use os_heap_init to get proper base and end addresses.

Reply:
Okay